addons/website_sale_coupon/wizard/sale_coupon_share.py

```diff
@@ -0,0 +1,113 @@
+    # -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, api, _
+from odoo.exceptions import UserError, ValidationError
+
+from werkzeug.urls import url_encode
+
+class SaleCouponShare(models.TransientModel):
+    _name = 'coupon.share'
+    _description = 'Create links that apply a coupon and redirect to a specific page'
+
+    website_id = fields.Many2one('website', required=True)
+    coupon_id = fields.Many2one('coupon.coupon', domain="[('program_id', '=', program_id)]")
+    program_id = fields.Many2one('coupon.program', required=True, domain=['|',
+        ('program_type', '=', 'coupon_program'),
+        ('promo_code_usage', '=', 'code_needed'),
+    ])
+    program_website_id = fields.Many2one('website', string='Program Website', related='program_id.website_id')
+    create_link_tracker = fields.Boolean(default=False)
+    link_tracker_id = fields.Many2one('link.tracker', compute='_get_share_link')
+
+    promo_code = fields.Char(compute='_get_promo_code')
+    share_link = fields.Char(compute='_get_share_link')
+    redirect = fields.Char(required=True)
+
+    related_link_ids = fields.One2many('link.tracker', compute='_get_related_links', inverse='_set_related_links')
+    related_link_count = fields.Integer(compute='_get_related_links')
```

Comment:
what is this used for? I only grep 2 times this field, here and in the compute that set its value?